extension/demo/src/bg/background.js

```diff
@@ -23,19 +24,34 @@ function onToken (data, sendResponse) {
   return true
 }
 
-chrome.storage.local.get(
-  [storageKey],
-  result => {
-    if (!result[storageKey]) {
-      openAuthTab()
-    }
+function timeToLive (expiresAt) {
+  return expiresAt - (Date.now() + 3600)
+}
+
+async function updatePayload () {
+  const payload = await api.fetchPayload()
+  if (payload) { // fetch return might need to go through .json()?
+    storage.store(keys.payload, payload)
   }
-)
+}
+
+function init () {
+  chrome.storage.local.get(
+    [keys.auth],
+    result => {
+      if (!result[keys.auth]) {
+        openAuthTab()
+      } else {
+        setTimeout(updatePayload, timeToLive(result.payload.expiresAt))
```

Comment:
Shouldn't this be done after the token is received as well?

Reply:
yes indeed